Instruments/nmr_signal_gui.py

```diff
@@ -120,24 +124,57 @@ def on_pick(self, event):
         QMessageBox.information(self, "Click!", msg)
 
     def set_field_conditional(self, Field):
+        I_setting = Field * self.myconfig["current_v_field_A_G"]
+        true_B0_G = self.h.field.to("T").magnitude * 1e4
+        print(
+            "adjusting current_v_field_A_G from",
+            self.myconfig["current_v_field_A_G"],
+        )
+        self.myconfig["current_v_field_A_G"] *= self.prev_field / true_B0_G
```

Comment:
this has the potential to error out -- note that the code below is checking that prev_field actually exists first.